### PR TITLE
Refine zorder plan

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

- Change origin `OptimizeZorderCommand` to `OptimizeZorderStatement` which holds the parsed plan
- Add `OptimizeZorderCommand` which delegate an another command(e.g. `InsertIntoHive`) to execute; the reason we should add a new command (not reuse `InserIntoStatement`) is we may need to implement read and overwrite same table in future. And also it's tricky to add `InserIntoStatement` during analyze.
- Change `ZorderBeforeWrite` to `ResolveZorder`, so we can resolve `OptimizeZorderStatement` to `OptimizeZorderCommand`

So the code should look cleaner

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: dev/kyuubi-extension-spark-3-1/pom.xml [hunk 1]
```diff
@@ -32,6 +32,12 @@
     <name>Kyuubi Project Dev Spark Extensions (for Spark 3.1)</name>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.kyuubi</groupId>
+            <artifactId>kyuubi-common_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (ulysses-you) [inline dev/kyuubi-extension-spark-3-1/pom.xml:39]: this is for `KyuubiSQLException`

**Comment by ulysses-you** [comment]
cc @qphien @hzxiongyinke @cfmcgrady @yaooqinn @pan3793

File: dev/kyuubi-extension-spark-3-1/pom.xml (hunk 1 above)
Comment (ulysses-you) [reply dev/kyuubi-extension-spark-3-1/pom.xml:39]: ah, seems we need to define a new exception in extension.

Comment (pan3793) [reply dev/kyuubi-extension-spark-3-1/pom.xml:39]: Yeah, decouple it from kyuubi-common will make life simple :)

**Review by cfmcgrady — APPROVED** [review]
LGTM

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks for review, merged to master

**Pull request closed without merging**